### PR TITLE
[Ticket/14203]Modularize and fix paging styling

### PR DESCRIPTION
**Pull request by hanakin**
ticket: https://tracker.phpbb.com/browse/PHPBB3-14203
PHPBB3-14203

### CODE REVIEW

**Comment by hanakin** [comment]
!set Styles

**Comment by bantu** [comment]
@hanakin I am sorry to inform you that we still need the PHPBB3-12345 id in each commit message as per https://wiki.phpbb.com/Git#Commit_Messages. You (or one of us) will have to adjust all your commits.

**Comment by hanakin** [comment]
@bantu can you clear up what this is this even used for as it seems sort of superfluous as we already have the ticket id in the message? I will try and refactor all my open PRs, but it just seems this could easily be factored out as a requirement if its a hook using some sort of global for the "PHPBB3"  and the "12345" from [ticket/12345].

**Comment by bantu** [comment]
@hanakin The things that matters for pinning commits to tracker tickets is the PHPBB3-12345 part. The message part is less important. Using "[feature/somefeaturename] Bla bla foo bar." is fine for the top of the message as well. There used to be a check on Travis CI ensuring all commits are well formatted. I am not sure whether it has been disabled or whether it is buggy.

**Comment by hanakin** [comment]
!set WIP

**Comment by hanakin** [comment]
!unset WIP

**Comment by hanakin** [comment]
!unset WIP

**Comment by hanakin** [comment]
!unset WIP

File: phpBB/styles/prosilver/theme/colours.css [hunk 1]
```diff
@@ -640,30 +640,18 @@ Colours and backgrounds for buttons.css
...
-	background: #4692BF;
-	border-color: #4692BF;
-	color: #FFFFFF;
+	color: #8f8f8f;
 }
 
-.pagination li a:hover, .pagination li a:hover .icon, .pagination .dropdown-visible a.dropdown-trigger, .nojs .pagination .dropdown-container:hover a.dropdown-trigger {
+.button-paging:hover,
+.button-paging:focus,
+.button-active {
```
Comment (prototech) [inline phpBB/styles/prosilver/theme/colours.css]: Missing the last selectors whose purpose is to keep the page jump button active while the dropdown is open.

**Comment by prototech** [comment]
The next/previous/page jump icons are not vertically aligned to the center. They seem about 2px too high.

File: phpBB/styles/prosilver/theme/colours.css (hunk 1 above)
Comment (hanakin) [inline phpBB/styles/prosilver/theme/colours.css]: needs addressed but not all of it is necessary ill get to it tommorow

**Comment by hanakin** [comment]
what browser/OS is it appearing offset i do not see it offset!

**Comment by prototech** [comment]
Ubuntu/Chromium 
pagination

**Comment by hanakin** [comment]
ready for review @VSEphpbb @cyberalien @prototech

**Comment by hanakin** [comment]
@prototech similar to https://github.com/phpbb/phpbb/pull/3937 this is to simplify the use of buttons and standardize the code base. It also makes working with icons easier

**Comment by hanakin** [comment]
@prototech ill let you decide on this one it has a lot of merit for making the paging easier to work with and a modular class to use elsewhere but per our discussion there are some BC issues

**Comment by Nicofuma** [comment]
@prototech what about the BC issue? The alpha2 is coming soon and after hat it will be too late (or almost) to introduce any new BC break.

**Comment by hanakin** [comment]
!set WIP

**Pull request closed without merging**